src/Battery.vala

```diff
@@ -60,10 +203,54 @@ namespace Power {
                     }
                 }
             } catch (Error e) {
-                critical ("acpi couldn't get upower devices");
+                critical ("ACPI couldn't get UPower devices");
             }
 
             return path;
         }
+
+        public string get_info () {
+            var percent = (int)Math.round (percentage);
+            if (percent <= 0) {
+                return _("Calculating…");
+            }
+
+            if (percent == 100) {
+                return _("Fully charged");
+            }
+
+            var info = _("%i%% charged").printf (percent);
+
+            return info;
+        }
+
+        public string get_health () {
+            var capacity = (int)Math.round (capacity);
+            if (capacity <= 60) {
+                return _("Poor");
+            }
+
+            if (capacity > 60) {
+                return _("Normal");
+            }
+
+            if (capacity > 90) {
+                return _("Excelent");
+            }
+
+            return _("Unknown");
+        }
```

Comment:
I spent some more time researching and thinking about this. I think it would be helpful to add a couple more levels and tweak the percentages we're using:

```suggestion
        public string get_health () {
            var capacity = (int)Math.round (capacity);

            if (capacity < 60) {
                return _("Critical");
            }

            if (capacity < 70) {
                return _("Poor");
            }

            if (capacity < 80) {
                return _("Fair");
            }

            if (capacity < 90) {
                return _("Good");
            }

            if (capacity >= 90) {
                return _("Excellent");
            }

            return _("Unknown");
        }
```